src/vault/secret/transit.clj

```diff
@@ -141,41 +146,68 @@
       not all members of a batch fail to encrypt.")
 
   (decrypt-data!
-    [client key-name ciphertext]
-    [client key-name ciphertext opts]
-    "Decrypt the ciphertext using the named key. Returns the plaintext string.
+    [client key-name data]
+    [client key-name data opts]
+    "Decrypt data using the named key.
+
+    In single-item mode, `data` should be the ciphertext string returned from
+    [[encrypt-data!]]. Returns a map with the `:plaintext` data decoded into a
+    byte array.
+
+    In batch mode, `data` should be a sequence of maps, each containing their
+    own `:ciphertext` and optional `:context`, `:nonce`, and `:reference`
+    entries. Returns a vector of batch results, each with `:plaintext` and
+    `:reference`.
 
     Options:
 
+    - `:as-string` (boolean)
+
+      Set to true to have the plaintext data decoded into a string instead of a
+      byte array.
+
     - `:associated-data` (string)
 
       Associated data to be authenticated (but not decrypted).
 
-    - `:context` (string)
+    - `:context` (string or bytes)
 
       The context for key derivation. Required if key derivation is enabled.
-      Must be sent as base64.
+      Automatically base64-encoded.
 
     - `:nonce` (bytes)
 
-      The nonce used for encryption. Must be sent as base64.
+      The nonce used for encryption. Automatically base64-encoded.
 
     - `:reference` (string)
 
-      A string to help identify results when using `:batch-input`.
-
-    - `:batch-input` (coll)
-
-      Specifies a list of items to decrypt in a single batch. Output preserves
-      order of input. Will ignore top-level `:ciphertext` and related fields if
-      set.
+      A string to help identify results when using batch mode. No effect in
+      single-item mode.
 
     - `:partial-failure-response-code` (integer)
 
       If set, will return this HTTP response code instead of a 400 if some but
       not all members of a batch fail to decrypt."))
 
 
+;; ## HTTP Client
+
+(defn- parse-key-info
+  "Parse the key information map returned by [[read-key]] and [[rotate-key!]]."
+  [body]
+  (let [data (u/kebabify-body-data body)
+        versions (into {}
+                       (map (fn parse-version
+                              [[version created-at]]
+                              [(or (parse-long (name version)) version)
+                               (try
+                                 (Instant/ofEpochSecond created-at)
+                                 (catch Exception _
+                                   created-at))]))
```

Comment:
When would `version` and `created-at` not be parseable?

Reply:
No idea, just trying to be defensive so that a caller which doesn't care about the key versions would still be able to use the rest of the information.